### PR TITLE
Don't restart requests when CouchDB returns an error.

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

medic/medic-webapp#4185

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Another day, another bug(fix?). Penny for your thoughts, @garethbowen . Thanks!

File: tests/e2e/api/controllers/changes.js [hunk 1]
```diff
@@ -753,6 +753,14 @@ describe('changes handler', () => {
           expect(changes.results.every(change => bobsIds.indexOf(change.id) !== -1 || change.id === 'org.couchdb.user:steve') ).toBe(true);
         });
     });
+
+    it('sends an error when couchdb returns an error', () => {
+      return requestChanges('bob', { style: 'couchdb will love this', seq_interval: 'this as well' })
+        .catch(err => {
+          expect(err).toBeTruthy();
+          expect(err.message.indexOf('Error processing your changes')).not.toEqual(-1);
```
Comment (garethbowen) [inline tests/e2e/api/controllers/changes.js]: Consider using `err.message.includes(...)` for readability.

File: api/src/controllers/changes.js [hunk 2]
```diff
@@ -164,6 +172,9 @@ const writeDownstream = (feed, content, end) => {
   if (feed.res.finished) {
     return;
   }
+  if (feed.error) {
+    feed.res.status(500);
```
Comment (garethbowen) [inline api/src/controllers/changes.js]: Is there a chance we've already sent headers?

Comment (dianabarsan) [reply api/src/controllers/changes.js]: Good point. Thanks!

**Comment by dianabarsan** [comment]
I've added a check for sent headers (thanks again!)

There's a slight problem. Because of the heartbeats we're sending, headers may be indeed already sent before we receive an error from CouchDB. So, errored requests which have already sent a heartbeat will have an error body but a 200 status. PouchDB only checks for the status, so it will consider the request successful. 

Continuous replication would not be affected, as PouchDB will keep requesting with the same seq, but initial replication fails: PouchDB considers the replication successful the app tries to go further, and redirects the user to the app /#error page, which might be confusing.

**Pull request merged**